components/autocomplete/completer-compat.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * This mod
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { deprecated } from '@wordpress/utils';
+
+const generateCompleterName = ( () => {
+	let count = 0;
+	return () => `backcompat-completer-${ count++ }`;
+} )();
+
+export function isDeprecatedCompleter( completer ) {
+	return 'onSelect' in completer;
+}
+
+export function toCompatibleCompleter( deprecatedCompleter ) {
+	deprecated( 'Original autocompleter interface', {
+		version: '2.8',
+		alternative: 'Latest autocompleter interface',
+		plugin: 'Gutenberg',
+		link: 'https://github.com/WordPress/gutenberg/blob/master/components/autocomplete/README.md',
+	} );
+
+	const compatibleCompleter = {
+		name: generateCompleterName(),
+		triggerPrefix: deprecatedCompleter.triggerPrefix,
+
+		options() {
+			return deprecatedCompleter.getOptions();
+		},
+
+		getOptionLabel( option ) {
+			return option.label;
+		},
+
+		getOptionKeywords( option ) {
+			return option.keywords;
+		},
+
+		getOptionCompletion() {
+			return {
+				action: 'backcompat',
+				value: deprecatedCompleter.onSelect.bind( deprecatedCompleter ),
+			};
+		},
+	};
+
+	if ( 'className' in deprecatedCompleter ) {
+		compatibleCompleter.className = deprecatedCompleter.className;
+	}
+	if ( 'allowNode' in deprecatedCompleter ) {
+		compatibleCompleter.allowNode = ( ...args ) => deprecatedCompleter.allowNode( ...args );
```

Comment:
Could you map other props like this?
```js
const extraProps = [ 'className', 'allowNode', 'allowContext' ].
    filter( key => key in deprecatedCompleter ).
    map( key => deprecatedCompleter[ key ];

const compatibleCompleter = {
    name: generateCompleterName(),
    // ...
    ...extraProps,
};
```
It seems like those functions don't add any new logic.

Reply:
Yep, that's more concise. I wanted to preserve completer context for the methods in that list, but `Autocomplete` already doesn't call them with completer context.